.github/workflows/deploy-project.yaml

```diff
@@ -78,76 +79,73 @@ jobs:
+      # This step uses the test-reporter action to publish test results in the workflow run #[[test-report-publisher]]
+      # It looks for any file named "TEST-...xml" in the project and uses it to publish the test results
+      # The job name in GitHub with the test results is "JUnit Test Results"
       - name: Publish Test Report
         uses: dorny/test-reporter@v1
         # If defines conditions that need to be fulfilled to run this step, in this case this step will always run
+        # That is because the report should be published, even if the tests failed
         if: always()
         with:
-          # Name for the report
-          name: JUnit Tests
-          # Where to look for report files, in this case look for any file named "TEST-...xml" anywhere in the project
-          path: ./**/TEST-*.xml
-          # What reporter created to use, since this project uses JUnit java-junit is used
-          reporter: java-junit
-          # Fail this step if there are test errors
-          fail-on-error: true
-          # Fail this step if no test results were found
-          fail-on-empty: true
+          name: JUnit Test Results # Name for the report
+          path: ./**/TEST-*.xml # Where to look for report files, any file named "TEST-...xml" anywhere in the project
+          reporter: java-junit # What reporter to use, since this project uses JUnit, the reporter is "java-junit"
+          fail-on-error: true # Fail this step if there are test errors
+          fail-on-empty: true # Fail this step if no test results were found
 
-      # Last step will upload the JAR file as an artifact that can then be used in other jobs
+      # This step uploads the JAR file as an artifact which can then be used in other jobs
       - name: Upload the JAR File
         uses: actions/upload-artifact@v4
         with:
-          # Artifact ID
-          name: packaged-project
-          # Where to find artifact file(s)
-          # Add "project.build.finalName --> ${project.artifactId}" to pom.xml to drop the version suffix
-          path: target/wls.jar
-          # Fail if artifact is not found
-          if-no-files-found: error
-          # Allow overwriting of previous artifacts
-          overwrite: true
+          name: packaged-project # Artifact ID, used to reference the artifact in other jobs when downloading it #[[packaged-project-id]]
+          path: target/wls.jar  # File(s) to include, the path is relative to the repository root
+                                #     Set "project.build.finalName --> ${project.artifactId}" in pom.xml to drop the version suffix
+          if-no-files-found: error # Fail if artifact is not found
+          overwrite: true # Allow overwriting of previous artifacts
 
-  # This job will build a Docker image with relevant tags and labels, and publish it in Harbor
+  # This job build a Docker image with relevant tags and labels, and publish it in Harbor
```

Comment:
Should be "This job build**s** a Docker image with relevant tags and labels, and publish it **to** Harbor"